content/posts/webpack/index.md

```diff
@@ -0,0 +1,91 @@
+---
+date: "2015-07-07"
+title: Webpack
+tags:
+  - javascript
+  - task-runner
+authors:
+  - MoOx
+header:
+  linearGradient: 160deg, rgb(204, 51, 51), rgba(204, 51, 51, .6)
+---
+
+# Pourquoi webpack ?
+
+Avant webpack, nous avions beau avoir des super task-runners comme
+[grunt](/posts/js/premiers-pas-avec-grunt)
+ou
+[gulp](/posts/js/introduction-gulp),
+il faut avouer qu'on bricolait quand même pas mal.
+
+Pourquoi ? Devoir ajuster des configurations/définitions de tâches pour ajuster
+les copies des images, fonts... Ajuster les urls de générations...
+Il reste avec ces solutions une grande partie de rafistolage, où l'on se doit de
+faire très attention à la moindre réorganisation de code, sous peine de casser
+une partie du rendu.
+
+Prenons par exemple une image de fond déclaré en CSS. Qui vous préviendra qu'une
+référence n'est pas bonne (à part peut-être vos logs de serveur web en
+recherchant les pages d'erreurs 404) ?
+
+Vous avez la responsabilité de gérer vos tâches et leurs résultats tout en vous
+assurant du bon fonctionnement car vous êtes la glu entre tous ces morceaux.
+
+# Webpack: faire un pack prêt pour le web
+
+Pas besoin de s'appeler Enstein pour comprendre les intentions de webpack en
+interprétant le nom: webpack.
+
+Webpack, bien que principalement orienté vers les modules JavaScript, va
+s'occuper de toutes les ressources que vous pouvez avoir besoin.
+Il est assez flexible pour consommer tous types de fichiers et en faire ce que
+vous voulez.
+
+En plus de cela, il a été pensé afin de permettre la séparation de votre pack
+en plusieurs morceaux, selon vos besoins: vous pourrez ainsi ajuster la balance
+entre performances et lazy loading.
+
+Avec les solutions existantes, il n'est actuellement pas possible d'arriver à
+ce que propose webpack.
+
+Webpack permet:
+
+- d'avoir toutes les ressources statiques en temps que module,
+- d'intégrer des bibliothèques tierces en tant que module,
+- de séparer votre "build" en plusieurs morceaux, chargé à la demande,
+- de garder un chargement initial rapide si besoin,
+- de personnaliser la plupart des étapes du processus,
+- de s'adapter aux gros projets.
+
+Beaucoup de comparatifs essaye de comparer
+[browserify](/posts/js/browserify-all-the-things) à
+webpack cela n'est pas nécessairement pertinent puisque webpack est beaucoup
+plus flexible que browserify de part sa conception et ses objectifs.
+
+Nativement, webpack ne peut s'occuper que des ressources JavaScript.
+Les *loaders* permettent de transformer tout et n'importe quoi en JavaScript
+(mais pas que). Ainsi, tout est consommable en tant que module.
+
+C'est cette partie qui est aujourd'hui la plus intéressante et la plus flexible.
+
+<figure>
+  ![](index.jpg)
+  <figcaption>
+    webpack transforme une multitude de fichier en lots par responsabilité
+  </figcaption>
+</figure>
+
+browserify ne va par exemple pas pouvoir appliquer des éventuelles
+transformations qu'au code local (donc pas dans `node_modules/*`).
```

Comment:
> Si c'est pas le cas, je dois pouvoir jeter ces posts et ragequit :) 
On se détend, l'article est bien. Ce qui me gène c'est le comparatif avec browserify. 
Les arguments que tu donnes :
> webpack est _beaucoup_ plus flexible que browserify
le system de transform de browserify c'est pas flexible ? 
> Les *loaders* permettent de transformer tout et n'importe quoi en JavaScript (mais pas que)
comme pour browserify
> browserify ne va par exemple pas pouvoir appliquer des éventuelles transformations qu'au code local 
Ahhh, c'est le seul argument contre et dont la tournure est complexe (un lien avec plus de détails serait top)

Du coup non, je trouve que le comparatif(pas l'article) ne réponds pas a ces questions.

En fait, plus j'y réfléchi et plus je me dit que je ferais juste un lien sur les 2 articles
https://github.com/webpack/docs/wiki/webpack-for-browserify-users & https://gist.github.com/substack/68f8d502be42d5cd4942

Après si je suis le seul que ça genre, oublie mes remarques.